config/index.ts

```diff
@@ -1,268 +1,405 @@
+    config.accounts,
+    updatedAccount.accountId
+  );
+
+  if (accountIndex < 0) {
+    throw new Error(
+      i18n('config.updateConfigAccount.accountNotFound', {
+        accountId: updatedAccount.accountId,
+      })
+    );
   }
+
+  config.accounts[accountIndex] = updatedAccount;
+
+  writeConfigFile(config, getConfigFilePath());
 }
 
-export function deleteConfigFile(): void {
-  if (CLIConfiguration.isActive()) {
-    newDeleteConfigFile();
-  } else {
-    config_DEPRECATED.deleteConfigFile();
+export function setConfigAccountAsDefault(identifier: number | string): void {
+  const config = getConfig();
+
+  const account = getConfigAccountByInferredIdentifier(
+    config.accounts,
+    identifier
+  );
+
+  if (!account) {
+    throw new Error(
+      i18n('config.setConfigAccountAsDefault.accountNotFound', {
+        accountId: identifier,
+      })
+    );
   }
+
+  config.defaultAccount = account.accountId;
+  writeConfigFile(config, getConfigFilePath());
 }
 
-export function isConfigFlagEnabled(flag: keyof CLIConfig): boolean {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.isConfigFlagEnabled(flag);
+export function renameConfigAccount(
+  currentName: string,
+  newName: string
+): void {
+  const config = getConfig();
+
+  const account = getConfigAccountByIdentifier(
+    config.accounts,
+    'name',
+    currentName
+  );
+
+  if (!account) {
+    throw new Error(
+      i18n('config.renameConfigAccount.accountNotFound', {
+        currentName,
+      })
+    );
   }
-  return config_DEPRECATED.isConfigFlagEnabled(flag);
-}
 
-export function isTrackingAllowed() {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.isTrackingAllowed();
+  const duplicateAccount = getConfigAccountByIdentifier(
+    config.accounts,
+    'name',
+    newName
+  );
+
+  if (duplicateAccount) {
+    throw new Error(
+      i18n('config.renameConfigAccount.duplicateAccount', {
+        newName,
+      })
+    );
   }
-  return config_DEPRECATED.isTrackingAllowed();
+
+  account.name = newName;
+
+  writeConfigFile(config, getConfigFilePath());
 }
 
-export function getEnv(nameOrId?: string | number) {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.getEnv(nameOrId);
+export function removeAccountFromConfig(accountId: number): void {
+  const config = getConfig();
+
+  const index = getConfigAccountIndexById(config.accounts, accountId);
+
+  if (index < 0) {
+    throw new Error(
+      i18n('config.removeAccountFromConfig.accountNotFound', {
+        accountId,
+      })
+    );
   }
-  return config_DEPRECATED.getEnv(nameOrId);
-}
 
-export function getAccountType(
-  accountType?: AccountType,
-  sandboxAccountType?: string | null
-): AccountType {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.getAccountType(accountType, sandboxAccountType);
+  config.accounts.splice(index, 1);
+
+  if (config.defaultAccount === accountId) {
+    delete config.defaultAccount;
```

Comment:
need to double check but I believe detecting this and prompting you to set a new default is handled by the CLI